### PR TITLE
Improve push and SMS messages.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR improves the content of push and SMS notifications so it fits in one push notification, and maybe two SMS messages at most. This involves placing the trip name at the top of the notification if no reminder is used, and summarizing the alerts instead of attempting to print the entire contents of alerts.

### CODE REVIEW

**Review by JymDyerIBI — APPROVED** [review]
👍  These look good.

Two unit tests fail on JDK 20 due to short time format changes in that JDK. However, I have already written fix for that under a separate ticket (OTP-983) which I will submit after these changes are committed/merged.

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java [hunk 1]
```diff
@@ -157,6 +158,35 @@ private static String bodyFromAlerts(
         return body.toString();
     }
 
+    private static String bodyShortFromAlerts(
+        Set<LocalizedAlert> previousAlerts,
+        Set<LocalizedAlert> resolvedAlerts,
+        Set<LocalizedAlert> unseenAlerts)
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java]: Nit. Should this closing bracket be on the next line?

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 2]
```diff
@@ -421,23 +424,30 @@ private void sendNotifications() {
         // Update push notification devices count, which may change asynchronously
         NotificationUtils.updatePushDevices(otpUser);
 
-        if (notifications.size() == 0) {
+        boolean hasInitialReminder = initialReminderNotification != null;
+
+        if (notifications.size() == 0 && !hasInitialReminder) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Could replace with `notifications.isEmpty()`

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 3]
```diff
@@ -421,23 +424,30 @@ private void sendNotifications() {
...
             // FIXME: Change log level
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:431]: Maybe out of scope, but can this `FIXME` be addressed in this PR?

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 4]
```diff
@@ -421,23 +424,30 @@ private void sendNotifications() {
...
             LOG.info("No notifications queued for trip. Skipping notify.");
             return;
         }
         // If the same notifications were just sent, there is no need to send the same notification.
         // TODO: Should there be some time threshold check here based on lastNotificationTime?
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:436]: Probably out of scope, but can this `TODO` be addressed in this PR?

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 5]
```diff
@@ -421,23 +424,30 @@ private void sendNotifications() {
...
             "tripId", trip.id,
-            "tripName", trip.tripName,
+            "tripNameOrReminder", tripNameOrReminder,
             "notifications", new ArrayList<>(notifications)
-        );
+        ));
+        if (hasInitialReminder) {
+            templateData.put("initialReminder", initialReminderNotification);
+        }
         // FIXME: Change log level
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:452]: Maybe out of scope, but can this FIXME be addressed in this PR?

**Review by br648 — APPROVED** [review]
Some minor items, but approving.

**Review by JymDyerIBI — APPROVED** [review]
Merge changes look good.

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java [hunk 6]
```diff
@@ -27,7 +27,7 @@ public TripMonitorAlertNotification(
         TripMonitorAlertSubNotification newAlertsNotification,
         TripMonitorAlertSubNotification resolvedAlertsNotification
     ) {
-        super(NotificationType.ALERT_FOUND, getBody(newAlertsNotification, resolvedAlertsNotification));
+        super(NotificationType.ALERT_FOUND, getBody(newAlertsNotification, resolvedAlertsNotification), getBodyShort(newAlertsNotification, resolvedAlertsNotification));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java]: Nit. Wrap line.

**Review by br648 — APPROVED** [review]
Just one minor comment and I assume my comments in `CheckMonitoredTrip` are out of scope? :-)

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:431]: It does warrant a review of trip monitoring logs, but that should be outside of this PR.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Fixed in https://github.com/ibi-group/otp-middleware/pull/205/commits/b5d2fce5f360398be8caabd60ffa7a4bff102068.

**Comment by binh-dam-ibigroup** [comment]
The merging with dev was quite significant, so I'm sending this PR back to review.

**Review by JymDyerIBI — APPROVED** [review]
👍

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java [hunk 7]
```diff
@@ -53,39 +56,64 @@ public static TripMonitorAlertNotification createAlertNotification(
...
-            body.append(newAlertsNotification);
-            body.append(System.lineSeparator());
-        }
-        if (resolvedAlertsNotification != null) {
-            body.append(resolvedAlertsNotification.toString(true));
+        // If there are any unseen or resolved alerts, notify accordingly.
+        boolean hasNewAlerts = !newAlerts.isEmpty();
+        boolean hasResolvedAlerts = !resolvedAlerts.isEmpty();
+        if (isAllClear) {
+            body.append("☑ All clear! All alerts on your itinerary were all resolved.");
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java]: Replace icon with `RESOLVED_ALERT_ICON`

File: src/test/java/org/opentripplanner/middleware/models/TripMonitorAlertNotificationTest.java
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/models/TripMonitorAlertNotificationTest.java:1]: Could use RESOLVED_ALERT_ICON and NEW_ALERT_ICON in these unit tests.

**Review by br648 — APPROVED** [review]
A couple of minor comments, else all good!

**Comment by binh-dam-ibigroup** [comment]
Appreciate the reviews!

File: src/test/java/org/opentripplanner/middleware/models/TripMonitorAlertNotificationTest.java
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/models/TripMonitorAlertNotificationTest.java:1]: Refactored in https://github.com/ibi-group/otp-middleware/pull/205/commits/1d8679e83d0b7b20c6fa157d54bea004d9f10b67.

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java (hunk 7 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/TripMonitorAlertNotification.java]: Refactored in https://github.com/ibi-group/otp-middleware/pull/205/commits/1d8679e83d0b7b20c6fa157d54bea004d9f10b67.

**Pull request merged**